addons/mail/static/src/core/common/thread.js

```diff
@@ -126,6 +132,34 @@ export class Thread extends Component {
                 },
             });
         }
+        if (this.env.inChatter && this.props.scrollRef) {
+            useScrollSnapshot(this.props.scrollRef, {
+                onWillPatch: () => {
+                    return {
+                        hasMoreMsgsAbove:
+                            this.props.thread.oldestPersistentMessage?.id <
+                                this.oldestPersistentMessage && this.props.order === "desc",
+                        hasMoreMsgBelow:
+                            this.props.thread.loadNewer &&
+                            this.props.thread.newestPersistentMessage?.id >
+                                this.newestPersistentMessage &&
+                            this.props.order === "desc",
+                    };
+                },
+                onPatched: ({ hasMoreMsgsAbove, hasMoreMsgBelow, scrollTop, scrollHeight }) => {
+                    const el = this.props.scrollRef.el;
+                    if (!this.isJumpingRecent && !this.env.messageHighlight?.highlightedMessageId) {
+                        if (hasMoreMsgsAbove) {
+                            el.scrollTop = scrollTop;
+                        } else if (hasMoreMsgBelow) {
+                            el.scrollTop = scrollTop + el.scrollHeight - scrollHeight;
+                        }
+                    }
+                    this.oldestPersistentMessage = this.props.thread.oldestPersistentMessage?.id;
+                    this.newestPersistentMessage = this.props.thread.newestPersistentMessage?.id;
+                },
+            });
```

Comment:
This is pretty much the same code as `useScrollSnapshot()` above when `!this.env.inChatter`, with `this.props.order` on `"desc"` instead of `"asc"`, using `scrollRef` instead of `messagesRef`.
The adjustment looks opposite of code above, but it's the same, just that `above`/`below` are inverted, which makes it quite hard to read.

It's best to keep a single `useScrollSnapshot`: reuse most of the logic as it should (above in asc => oldest, above in desc => newest. Right now it states above in desc => oldest, which is wrong)

I think the `ref` param can be a function instead, so that you can `useScrollSnapshot()` regardless of the config, and let function determine the appropriate ref.